dandi/models.py

```diff
@@ -113,6 +113,104 @@ def diff_models(model1, model2):
             print(f"{field} is different")
 
 
+def to_datacite(dandiset):
+    from .metadata import migrate2newschema
+
+    meta = dandiset.metadata
+    newmeta = migrate2newschema(meta)
+
+    dandiset_id = dandiset.identifier
+
+    attributes = {}
+    attributes["identifiers"] = [
+        {"identifier": newmeta.doi, "identifierType": "DOI"},
+        # not sure if I can add dandi_id in the PDF documentation the only option is DOI...
+        {"identifier": dandiset_id, "identifierType": "Dandi"},
+    ]
+    attributes["titles"] = [{"title": newmeta.name}]
+    attributes["descriptions"] = [
+        # not sure what should be the type os use Other
+        # options are: Abstract, Methods, SeriesInformation,
+        # TableOfContents, TechnicalInfo, Other
+        {"description": newmeta.description, "descriptionType": "Other"}
+    ]
+    attributes["publisher"] = "DANDI Archive"
+    attributes["publicationYear"] = datetime.now().year
+    # not sure about it dandi-api had "resourceTypeGeneral": "NWB"
+    attributes["types"] = {"resourceType": "NWB", "resourceTypeGeneral": "Dataset"}
+    # newmeta has also attribute url, but it often empty
+    attributes["url"] = newmeta.url
+    # assuming that all licenses are from SPDX?
+    attributes["rightsList"] = [
+        {
+            "schemeURI": "https://spdx.org/licenses/",
+            "rightsIdentifierScheme": "SPDX",
+            "rightsIdentifier": el.name,
+        }
```

Comment:
this looks good - i'm going to update the license types in dandi to directly use this.